src/Rule/Subset.php

```diff
@@ -4,34 +4,33 @@
 
 namespace Yiisoft\Validator\Rule;
 
+use Attribute;
 use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Validator\FormatterInterface;
 use Yiisoft\Validator\Result;
 use Yiisoft\Validator\Rule;
 use Yiisoft\Validator\ValidationContext;
 
+#[Attribute(Attribute::TARGET_PROPERTY)]
 final class Subset extends Rule
 {
-    /**
-     * @var iterable
-     */
-    private iterable $values;
-    /**
-     * @var bool whether the comparison is strict (both type and value must be the same)
-     */
-    private bool $strict = false;
-
-    private string $iterableMessage = 'Value must be iterable';
-
-    private string $subsetMessage = 'Values must be ones of {values}.';
-
-    public static function rule(iterable $values): self
-    {
-        $rule = new self();
-        $rule->values = $values;
-        return $rule;
+    public function __construct(
+        private iterable $values,
+        /**
+         * @var bool whether the comparison is strict (both type and value must be the same)
+         */
+        private bool $strict = false,
+        private string $iterableMessage = 'Value must be iterable',
```

Comment:
```suggestion
        private string $iterableMessage = 'Value must be iterable.',
```

Reply:
That's how it was before. But agree, let's fix it along the way.

Reply:
:heavy_check_mark: